### ISSUE TITLE
Make grepFilterSpecs work with grepOmitFiltered

### ISSUE THREAD

**Issue by bahmutov** [cypress-io/cypress-grep#85]
To pre-filter specs that only have tagged tests

We need a better structure of test tags - where we find all suites and tests with their tags

Blocked by https://github.com/bahmutov/find-test-names/issues/5

**Comment by melibe23** [comment]
I just want to say that I WOULD LOVE this enhancement.

**Comment by CoryDanielson** [comment]
@melibe23 I work around this issue when running via `yarn run` on CI by grepping test files and passing them in via `--spec` it's not bulletproof, but it'll be a huge win for you.

Our Jenkins job has parameters for `teamTags` and `suiteTags` which can be any one of these values
- **teamTags**: `"all" "@team1" "@team2"`
- **suiteTags**: `"all" "@nightly" "@smoke" "@critical" "@smoke @critical"`

There is a step in our job that grabs all of the filtered tags that are not "all" and saves them to an array (splitting by space for `@smoke @critical`). So the array can end up looking like this:
- `["@team1", "@smoke", "@critical"]`

With that array, you can grep through your test files for those strings, as a rough but effective way to filter the tests. Any test file with those strings will most likely have the tags (unless you have an alias set up with a similar name (ie `cy.request(...).as("@criticalRequest")`)

To filter the tests, you can loop over the array and grep your test files like so (this is JS syntax sudo code)
```
// For the first grep, we'll search for all files in cypress/integration
const filteredTestFiles = testTagsToFilterBy.reduce((testPaths = "cypress/integration", tag) => {
    // This grep searches for all *.js files within filesToSearchIn, and then replaces all the new lines with a space
    // so that they are a space-separated list of files
    // you can then use that space-separated list as the filesToSearchIn argument for grep
    return sh("grep -lR --include='*.js' '${tag}' ${filesToSearchIn} | tr '\n' ' '");
})
```

The first grep will return all test files, and each following grep will operate on the results from the previous. At the end, you'll have a list of files that contain the desired tags. The greps will look like this:
 1. `grep @team1 cypress/integration`
 2. `grep @smoke cypress/integration/test1.test.js cypress/integration/test2.test.js  cypress/integration/test2.test.js` 
 3. `grep @critical cypress/integration/test1.test.js cypress/integration/test2.test.js`

We then pass the tags to cypressGrep via `cypress run --env grepTags="@smoke @critical"` when running the tests. They're space separated here so that we run all tests that contain any of the matched tags, and to keep it simple.

**Comment by Narretz** [comment]
Using the structure from find-test-name would also enable to pre-filter specs when using inverted filters, right?
Do you plan on working on this in the near future @bahmutov?
Otherwise it's something I might be able to work on.

**Comment by jordanpowell88** [comment]
We are in the process of deprecating this repo as this package has been republished under `@cypress/grep` and has moved to [https://github.com/cypress-io/cypress/tree/develop/npm/grep](https://github.com/cypress-io/cypress/tree/develop/npm/grep). I'm going to go ahead and close this issue as it appears stale. If this is still an issue, please open a new issue [here](https://github.com/cypress-io/cypress/issues/new/choose)
